lib/src/Base/Stat/ProcessSampleImplementation.cxx

```diff
@@ -351,6 +360,124 @@ ProcessSampleImplementation ProcessSampleImplementation::computeQuantilePerCompo
   return result;
 }
 
+/*
+ * Method computeQuantilePerComponent() gives the quantile per component of the sample using Robbins Monro approximation
+ */
+Sample ProcessSampleImplementation::computeQuantilePerComponentRobbinsMonro(const Scalar prob) const
+{
+  const UnsignedInteger size = getSize();
+  if (size == 0) return Sample();
+  if (size == 1) return data_[0];
+  const UnsignedInteger K = ResourceMap::GetAsUnsignedInteger("ProcessSample-RobbinsMonroProximalIterations");
+  const Scalar a1 = ResourceMap::GetAsScalar("ProcessSample-RobbinsMonroProximalFactor");
+  const Scalar gamma = ResourceMap::GetAsScalar("ProcessSample-RobbinsMonroExponent");
+  const UnsignedInteger dimension = data_[0].getDimension();
+  const UnsignedInteger length = data_[0].getSize();
+  const UnsignedInteger flatSize = length * dimension;
+  Point quantiles(data_[0].getImplementation()->getData());
+  for (UnsignedInteger i = 1; i < size; ++i)
+    {
+      const Point & Xi = data_[i].getImplementation()->getData();
+      const Scalar tau = 1.0 / pow(i + 1.0, gamma);
+      for (UnsignedInteger j = 0; j < flatSize; ++j)
+	{
+	  const Scalar theta0 = quantiles[j];
+	  const Scalar xij = Xi[j];
+	  Scalar thetaj = theta0;
+	  for (UnsignedInteger k = 0; k < K; ++k)
+	    {
```

Comment:
instead of setting dead code in ifdef, we could add some comments and highlight differences between "classical implementation" and the used one. It makes the code easy to read 
Also please mention the references as in PR msg